internal/gitinterface/blob_test.go

```diff
@@ -0,0 +1,88 @@
+package gitinterface
+
+import (
+	"os"
+	"os/exec"
+	"path"
+	"testing"
+
+	"github.com/adityasaky/gittuf/internal/common"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestReadBlob(t *testing.T) {
+	testDir, err := common.CreateTestRepository()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	readContents := []byte("test file read")
+	if err := os.WriteFile(path.Join(testDir, "test-file-read"), readContents, 0644); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := os.Chdir(testDir); err != nil {
+		t.Fatal(err)
+	}
+
+	cmd := exec.Command("git", "hash-object", "-w", "test-file-read")
+	if err := cmd.Run(); err != nil {
+		t.Fatal(err)
+	}
+
+	repo, err := common.GetRepositoryHandler()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Run("test expected file", func(t *testing.T) {
+		expectedHash := "2ecdd330475d93568ed27f717a84a7fe207d1c58"
+
+		length, contents, err := ReadBlob(repo, plumbing.NewHash(expectedHash))
+		if err != nil {
+			t.Error(err)
+		}
+
+		assert.Equal(t, len(readContents), length)
+		assert.Equal(t, readContents, contents)
+	})
+
+	t.Run("test non existent file", func(t *testing.T) {
+		length, _, err := ReadBlob(repo, plumbing.ZeroHash)
+		assert.ErrorIs(t, err, plumbing.ErrObjectNotFound)
+		assert.Equal(t, -1, length)
+	})
+
+}
+
+func TestWriteBlob(t *testing.T) {
+	testDir, err := common.CreateTestRepository()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	writeContents := []byte("test file write")
+	if err := os.WriteFile(path.Join(testDir, "test-file-write"), writeContents, 0644); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := os.Chdir(testDir); err != nil {
+		t.Fatal(err)
+	}
+
+	repo, err := common.GetRepositoryHandler()
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
```suggestion
	repo, err := git.Init(memory.NewStorage(), memfs.New())
	if err != nil {
		t.Fatal(err)
	}
```

See https://github.com/sigstore/gitsign/blob/2509c5c5f931b5de2cc9ca2449d7e309a65a2ec9/internal/config/config_test.go#L32-L35